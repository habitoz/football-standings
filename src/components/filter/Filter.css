.filterInput {
    border-radius: 10px;
    border: 1px solid #2E2E2C;
    background-color: #2E2E2C;
    width: 100%;
    padding: 10px 10px;
    margin-bottom: 20px;
    color: #fff;
}

.filterInput:focus {
    outline: none;
}