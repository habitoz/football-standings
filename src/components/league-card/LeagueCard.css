.CardContainer {
    width: 100%;
}

.LeagueImage {
    min-height: 9rem;
    width: 100%;
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center;
    background-color: #2E2E2C;
}

.LeagueName {
    text-align: center;
    padding: 7px 5px;
    background-color: #2E2E2C;
}