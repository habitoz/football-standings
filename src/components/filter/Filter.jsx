import React from 'react';
import PropTypes from 'prop-types';
import './Filter.css';

function Filter({ filterHandler }) {
  return (
    <div>
      <input className="filterInput" onChange={($e) => filterHandler($e.target.value)} type="text" />
    </div>
  );
}

Filter.propTypes = {
  filterHandler: PropTypes.func.isRequired,
};

export default Filter;
