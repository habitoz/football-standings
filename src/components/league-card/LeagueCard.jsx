import React from 'react';
import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { chengePage } from '../../redux/feature/app-info/appInfoSlice';
import './LeagueCard.css';

function LeagueCard({
  name, id, image,
}) {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const leagueSeasons = (id) => {
    navigate(`/seasons/${id}`);
    dispatch(chengePage('Seasons'));
  };
  return (
    <div className="CardContainer" onKeyUp={() => {}} role="navigation" onClick={() => leagueSeasons(id)}>
      <div style={{ backgroundImage: `url("${image}")` }} className="LeagueImage" />
      <p className="LeagueName">{name}</p>
    </div>
  );
}

LeagueCard.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};
export default LeagueCard;
