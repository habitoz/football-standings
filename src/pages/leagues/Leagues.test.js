import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom'; // Importamos MemoryRouter
import Filter from '../../components/filter/Filter';
import LeagueCard from '../../components/league-card/LeagueCard';

const mockStore = configureStore([]);

test('Leagues Page snapshot', () => {
  const initialState = {
    league: {
      leagues: [
        {
          id: 1,
          name: 'English PL',
          logos: { dark: '' },
        },
      ],
      searchString: 'United',
    },
  };
  const store = mockStore(initialState);

  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Filter filterHandler={() => {}} />
          <LeagueCard />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
