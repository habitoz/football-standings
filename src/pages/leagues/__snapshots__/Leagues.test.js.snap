// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leagues Page snapshot 1`] = `
Array [
  <div>
    <input
      className="filterInput"
      onChange={[Function]}
      type="text"
    />
  </div>,
  <div
    className="CardContainer"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="navigation"
  >
    <div
      className="LeagueImage"
      style={
        Object {
          "backgroundImage": "url(\\"undefined\\")",
        }
      }
    />
    <p
      className="LeagueName"
    />
  </div>,
]
`;
